### PR TITLE
Add new resource Global Component Contact

### PR DESCRIPTION
**Pull request by jacobseth**
This patch adds the global-component-contact resource, along with tests.

### CODE REVIEW

File: spec/fixtures/vcr/must_has_compose1.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
```
Comment (simozhan) [inline spec/fixtures/vcr/must_has_compose1.yml]: Is this file needed?

File: spec/pdc/v1/global_component_contact_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    VCR.insert_cassette fixture_name
+  end
+
+  after do
+    VCR.eject_cassette
+  end
+
+  let(:contact) { PDC::V1::GlobalComponentContact }
+
+  it 'must has compose' do
```
Comment (simozhan) [inline spec/pdc/v1/global_component_contact_spec.rb]: is this "must have component"

**Comment by ycheng-aa** [comment]
It has a little possibility be caused by token problem.  `pronto` will post comments just looks like they are posted by the token's owner. If the token is ineffective or wrong, this kind of failure will occur. Maybe some other reasons I don't know. Just provide a little information. 

I think the token might be configured as an environment variable in secret form in .travis.yml.

**Comment by ycheng-aa** [comment]
oh, I saw https://github.com/product-definition-center/pdc-ruby-gem/commit/9f277af34384297648eec83b0f1fd9900ecc2bcd . Looks .travis.yml is already removed the token part. Sorry for the misleading. I am not sure if pronto could work without an effective token. But looks it worked well previously?

**Comment by simozhan** [comment]
@ycheng-aa np, thanks for your active response :)

**Comment by ycheng-aa** [comment]
@simozhan  Is it possible that `pronto` can't work at all, but previous commits didn't have any code violation but this time had then `pronto` tried to post comment and error occurred? Just a wild guess.

File: examples/pdc_global_component_contact.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    config.disable_caching = true
+  end
+
+  comp_contact = PDC::V1::GlobalComponentContact.where(
+    component: 'vlgothic-fonts'
+  ).find_one!
+  puts comp_contact.contact
+end
+
+main if __FILE__ == $PROGRAM_NAME
```
Comment (xiangge) [inline examples/pdc_global_component_contact.rb]: There is a error occurred on travis when execute "bundle exec pronto run -f github_pr -c origin/${TRAVIS_BRANCH}".

> /home/travis/.rvm/gems/ruby-2.2.2/gems/octokit-4.7.0/lib/octokit/response/raise_error.rb:16:in `on_complete': POST https://api.github.com/repos/product-definition-center/pdc-ruby-gem/pulls/35/comments: 404 - Not Found // See: https://developer.github.com/v3/pulls/comments/#create-a-comment (Octokit::NotFound)

Seems remove this file the command can work, not know why yet.

**Comment by simozhan** [comment]
@xiangge Good catch! I used rubocop locally to have a test. There is  error "examples/pdc_global_component_contact.rb:18:9: C: Style/YodaCondition: Reverse the order of the operands __FILE__ == $PROGRAM_NAME. (https://en.wikipedia.org/wiki/Yoda_conditions)". After fixing this, this issue disappears. We have some of robocop errors, I guess some time before the TOKEN doesn't work which maybe involves these errors, but robocop runs only check the commit diff's error, so these existing errors will not be found in the next release. Let's fix that in another patch

**Comment by simozhan** [comment]
@jacobseth can you fix the robocop error and resubmit? Thanks!

**Comment by romanofski** [comment]
Yeah I think making the tests happy is a good thing, everything else seems fine to me.

**Comment by jacobseth** [comment]
Sorry for taking so long to reply. I was unsure whether this patch was still needed, so I went away to do some other work. I just pushed your recommended changes, thank-you for the reviews.

File: spec/pdc/v1/global_component_contact_spec.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
+require 'spec_helper'
+
+describe PDC do
```
Comment (simozhan) [inline spec/pdc/v1/global_component_contact_spec.rb]: I think this should be "describe PDC::V1::GlobalComponentContac" so that all these tests in the same label, like following:
PDC::V1::ContentDeliveryRepo::count
  test_0001_has serveral cdn content delivery repos               PASS (0.00s)

File: spec/pdc/v1/global_component_contact_spec.rb (hunk 2 above)
Comment (simozhan) [reply spec/pdc/v1/global_component_contact_spec.rb]: component contact doesn't have compose,
should be "must have component contact", and at meanwhile, please check the vcr file name accordingly.

File: spec/pdc/v1/global_component_contact_spec.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    it 'works with where' do
+      count = contact.where(component: 'vlgothic-fonts' ).count
+      count.must_equal 3 # as there is 3 fields per entry
+    end
+
+    it 'can find global contact by component name' do
+      count = contact.where(component: 'vlgothic-fonts').count
+      count.must_equal 3
+    end
+  end # count
```
Comment (simozhan) [inline spec/pdc/v1/global_component_contact_spec.rb]: let's remove # count as it will give pronto error, don't know why it doesn't pop up error in travis ci.

File: spec/pdc/v1/global_component_contact_spec.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    it 'returns a mail_name' do
+      vf = contact.where(component: 'vlgothic-fonts').first
+      vf.contact.mail_name.must_equal 'I18N'
+    end
+
+    it 'returns a gloabl contact id' do
+      vf = contact.where(component: 'vlgothic-fonts').first
+      vf.contact.id.must_equal 5
+    end
+  end # contact
```
Comment (simozhan) [inline spec/pdc/v1/global_component_contact_spec.rb]: same with above

File: spec/pdc/v1/global_component_contact_spec.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+  describe '#role' do
+    it 'returns contact role' do
+      bash_contact = contact.where(component: 'bash').first
+      bash_contact.role.wont_be_nil
+      bash_contact.role.must_equal 'QE_Group'
+    end
+  end # role
```
Comment (simozhan) [inline spec/pdc/v1/global_component_contact_spec.rb]: same with above, pronto error

File: spec/fixtures/vcr/works_with_where2.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
+---
```
Comment (simozhan) [inline spec/fixtures/vcr/works_with_where2.yml]: Is this file still needed?

File: spec/pdc/v1/global_component_contact_spec.rb [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  end
+
+  describe 'count' do
+    it 'must return number of resources' do
+      count = contact.count
+      count.must_equal 12546
+    end
+
+    it 'works with where' do
+      count = contact.where(component: 'vlgothic-fonts' ).count
```
Comment (simozhan) [inline spec/pdc/v1/global_component_contact_spec.rb]: space between ' )

File: spec/pdc/v1/global_component_contact_spec.rb (hunk 7 above)
Comment (romanofski) [reply spec/pdc/v1/global_component_contact_spec.rb]: Actually, out of curiosity, @simozhan where do you see the pronto error? I've checked the last(?) run:

https://travis-ci.org/product-definition-center/pdc-ruby-gem/builds/312237759?utm_source=github_status&utm_medium=notification

and couldn't find any errors.

Comment (simozhan) [reply spec/pdc/v1/global_component_contact_spec.rb]: @romanofski I test it locally with rubocop spec/pdc/v1/global_component_contact_spec.rb 
Yeah. You are right. There is no errors in the travis build. Actually I have the same question why travis not pop them out. These codes should be in the diff, why no errors

File: spec/pdc/v1/global_component_contact_spec.rb (hunk 4 above)
Comment (jacobseth) [reply spec/pdc/v1/global_component_contact_spec.rb]: Agreed

**Comment by jacobseth** [comment]
I have added the changes you recommended @simozhan.

File: examples/pdc_global_component_contact.rb [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env ruby
+
+require './lib/pdc'
+
+def main
+  PDC.configure do |config|
+    config.site = 'https://pdc.engineering.redhat.com'
```
Comment (romanofski) [inline examples/pdc_global_component_contact.rb]: Not that big of a deal, but since it's an upstream project, I'd use the Fedora PDC here.

Comment (jacobseth) [reply examples/pdc_global_component_contact.rb]: I agree @romanofski, however I think this belongs in its own PR, as the token fetch for this gem is set-up for this url.

Comment (xiangge) [reply examples/pdc_global_component_contact.rb]: @jacobseth yes, I think so, it's better to keep this url.
